tests/serialization/test_serde.py

```diff
@@ -309,3 +309,31 @@ def test_stringify(self):
     )
     def test_serialized_data(self, obj, expected):
         assert expected == serialize(obj)
+
+    def test_serialize_nested_primitive_and_non_primitive_values(self):
+        import pandas
+
+        def _compare(actual_obj, expected_obj):
+            assert len(actual_obj) == len(expected_obj)
+            for key in actual_obj:
+                if isinstance(actual_obj[key], dict):
+                    return _compare(actual_obj[key], expected_obj[key])
+                if isinstance(actual_obj[key], pandas.DataFrame):
+                    assert actual_obj[key].equals(expected_obj[key])
+                else:
+                    assert actual_obj[key] == expected_obj[key]
+
+        data = (
```

Comment:
why not use `@pytest.mark.parametrize` here, makes it consistent with other test cases.